crates/proto/src/rr/resource.rs

```diff
@@ -441,18 +441,12 @@ impl<'r> BinDecodable<'r> for Record {
                 ))
             })?;
 
-        // this is to handle updates, RFC 2136, which uses 0 to indicate certain aspects of
```

Comment:
lol, how did I miss this comment in the previous review... it explains why we allow NULL here: 

`this is to handle updates, RFC 2136, which uses 0 to indicate certain aspects of pre-requisites`

i.e. this is perfectly valid, and in-fact, required for Dynamic DNS. I believe a NULL rdata with a type in Dynamic DNS is a prereq check (or delete operation) for a name with any value for the RDATA section, for matching. So we'll need to rethink the fix for this.